### PR TITLE
Converts all JNI calls for Android wallet into mojo calls

### PR DESCRIPTION
**Pull request by SergeyZhukovsky**
Resolves https://github.com/brave/brave-browser/issues/17228

## Submitter Checklist:

- [ ] I confirm that no security/privacy review [is needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [ ] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [ ] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [ ] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [ ] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [ ] Checked the PR locally: `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests`, `npm run lint`, `npm run gn_check`, `npm run tslint`
- [ ] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

### CODE REVIEW

File: android/java/org/chromium/chrome/browser/crypto_wallet/fragments/onboarding_fragments/VerifyRecoveryPhraseFragment.java [hunk 1]
```diff
@@ -55,36 +60,61 @@ public View onCreateView(
...
+                        if (result.equals(recoveryPhraseToVerify)) {
+                            onNextPage.gotoNextPage(true);
+                            showAddAccountDialog();
+                        } else {
+                            phraseNotMatch();
+                        }
+                    });
+                } else {
+                    phraseNotMatch();
+                    ;
```
Comment (deeppandya) [inline android/java/org/chromium/chrome/browser/crypto_wallet/fragments/onboarding_fragments/VerifyRecoveryPhraseFragment.java:81]: Typo here for `;`

File: android/java/org/chromium/chrome/browser/crypto_wallet/AssetRatioControllerFactory.java [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+            if (instance == null) {
+                instance = new AssetRatioControllerFactory();
+            }
+        }
+        return instance;
+    }
+
+    private AssetRatioControllerFactory() {}
+
+    public AssetRatioController GetAssetRatioController(
```
Comment (deeppandya) [inline android/java/org/chromium/chrome/browser/crypto_wallet/AssetRatioControllerFactory.java:35]: In java, function names should be in camelcase.

File: android/java/org/chromium/chrome/browser/crypto_wallet/AssetRatioControllerFactory.java [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+        MessagePipeHandle handle = WrapNativeHandle(nativeHandle);
+        AssetRatioController assetRatioController =
+                AssetRatioController.MANAGER.attachProxy(handle, 0);
+        Handler handler = ((Interface.Proxy) assetRatioController).getProxyHandler();
+        handler.setErrorHandler(connectionErrorHandler);
+
+        return assetRatioController;
+    }
+
+    private MessagePipeHandle WrapNativeHandle(int nativeHandle) {
```
Comment (deeppandya) [inline android/java/org/chromium/chrome/browser/crypto_wallet/AssetRatioControllerFactory.java:48]: function names should be in camelcase.

**Comment by SergeyZhukovsky** [comment]
I put skip labels for everything but Android after the initial build passed all platforms. A commit https://github.com/brave/brave-core/pull/9584/commits/d22f34d6640b7b59042bdbffbef685e65bfc337c changed java files only, so we are good to verify Android only after that.

**Review by bbondy — APPROVED** [review]

File: android/java/org/chromium/chrome/browser/crypto_wallet/fragments/onboarding_fragments/SecurePasswordFragment.java [hunk 4]
```diff
@@ -64,16 +75,26 @@ public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceStat
...
             }
         });
     }
 
+    private void goToTheNextPage() {
+        String passwordInput = passwordEdittext.getText().toString().trim();
+        KeyringController keyringController = getKeyringController();
+        if (keyringController != null) {
+            keyringController.createWallet(passwordInput,
+                    recoveryPhrases
```
[the remark this answers is not in the document]
Comment (SergeyZhukovsky) [reply android/java/org/chromium/chrome/browser/crypto_wallet/fragments/onboarding_fragments/SecurePasswordFragment.java:89]: I don't think we need to do anything here

File: android/java/org/chromium/chrome/browser/crypto_wallet/activities/AssetDetailActivity.java [hunk 5]
```diff
@@ -96,6 +92,19 @@ public boolean onTouch(View v, MotionEvent event) {
         onInitialLayoutInflationComplete();
     }
 
+    private void getPriceHistory(String asset, int timeframe) {
+        if (mAssetRatioController != null) {
+            mAssetRatioController.getPriceHistory(asset, timeframe, (result, priceHistory) -> {
+                final float[] pricesArray = new float[priceHistory.length];
+                for (int index = 0; index < priceHistory.length; index++) {
+                    // TODO(sergz): We need to retrieve and pass date as well
```
[the remark this answers is not in the document]
Comment (SergeyZhukovsky) [reply android/java/org/chromium/chrome/browser/crypto_wallet/activities/AssetDetailActivity.java:100]: https://github.com/brave/brave-browser/issues/17241

**Review by deeppandya — APPROVED** [review]
LGTM

File: browser/brave_wallet/android/asset_ratio_controller_factory_android.cc [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+#include "brave/build/android/jni_headers/AssetRatioControllerFactory_jni.h"
+#include "chrome/browser/profiles/profile.h"
+#include "chrome/browser/profiles/profile_manager.h"
+#include "mojo/public/cpp/bindings/pending_receiver.h"
+
+namespace chrome {
+namespace android {
+static jint JNI_AssetRatioControllerFactory_GetInterfaceToAssetRatioController(
+    JNIEnv* env) {
+  auto* profile = ProfileManager::GetActiveUserProfile()->GetOriginalProfile();
```
Comment (bridiver) [inline browser/brave_wallet/android/asset_ratio_controller_factory_android.cc:17]: we should remove `GetOriginalProfile` here. I think this was a mistake carried over from the JNI code. We should always pass the actual profile (regular or incognito) and let the service factory decide what to do. We're circumventing that logic here with `GetOriginalProfile`

File: browser/brave_wallet/android/keyring_controller_factory_android.cc [hunk 7]
```diff
@@ -0,0 +1,26 @@
...
+#include "brave/build/android/jni_headers/KeyringControllerFactory_jni.h"
+#include "chrome/browser/profiles/profile.h"
+#include "chrome/browser/profiles/profile_manager.h"
+#include "mojo/public/cpp/bindings/pending_receiver.h"
+
+namespace chrome {
+namespace android {
+static jint JNI_KeyringControllerFactory_GetInterfaceToKeyringController(
+    JNIEnv* env) {
+  auto* profile = ProfileManager::GetActiveUserProfile()->GetOriginalProfile();
```
Comment (bridiver) [inline browser/brave_wallet/android/keyring_controller_factory_android.cc:17]: same as above

**Review by bridiver — APPROVED** [review]

**Pull request merged**